src/ghga_transpiler/metasheet_parser.py

```diff
@@ -0,0 +1,118 @@
+
+from openpyxl import Workbook
+from pydantic import BaseModel, Field
+
+from .config import WorkbookConfig
+from .exceptions import MetaColumnNotFound
+
+
+class MetaInfo(BaseModel):
+    """Class with constants that is requires to parse the configuration worksheets
+    of a workbook.
+    """
+
+    column_meta: str = Field(
+        default="__column_meta",
+        description="Name of a sheet that"
+        + " consists of column settings of the individual"
+        + " worksheets in a workbook.",
+    )
+    sheet_meta: str = Field(
+        default="__sheet_meta",
+        description="Name of a sheet that"
+        + " consists of general settings of individual worksheets"
+        + "(e.g. header_row, start_column) in a workbook.",
+    )
+    name_column: str = Field(
+        default="sheet",
+        description="The name of the column in"
+        + " COLUMN_META and SHEET_META worksheets that holds the"
+        + " names of the worksheets in the workbook that the settings"
+        + " are apply to.",
+    )
+
+
+def read_meta_information(workbook: Workbook, meta_sheet_name: str):
+    """Reads the content of a worksheet"""
+    if meta_sheet_name in workbook.sheetnames:
+        sheet_meta_header = [cell.value for cell in workbook[meta_sheet_name][1]]
+        sheet_meta_values = list(
+            workbook[meta_sheet_name].iter_rows(min_row=2, values_only=True)
+        )
+        return [
+            dict(zip(sheet_meta_header, val, strict=True)) for val in sheet_meta_values
+        ]
+    raise SyntaxError(
+        f"Unable to extract the sheet {meta_sheet_name} from the workbook."
+    )
+
+
+def reshape_columns_meta(column_meta: list, name_column: str) -> dict[str, list]:
+    """Reshapes column metadata into a dictionary where keys are worksheet
+    names and values are lists of column metadata dictionaries. Worksheet names comes
+    from the column 'name_column'.
+    """
+    worksheet_columns: dict = {}
+    for item in column_meta:
+        try:
+            sheet_name = item.get(name_column)
+        except KeyError as err:
+            raise MetaColumnNotFound(
+                f"{name_column} column with sheet is not found"
+            ) from err
+        worksheet_columns.setdefault(sheet_name, []).append(item)
+    return worksheet_columns
+
+
+def reshape_settings_meta(settings_meta: list, name_column: str) -> dict[str, dict]:
+    """Reshapes settings metadata into a dictionary where keys
+    are worksheet names and values are worksheet settings dictionaries.
+    Worksheet names comes from the column 'name_column'.
+    """
+    worksheet_settings: dict = {}
+    for item in settings_meta:
+        try:
+            sheet_name = item.get(name_column)
+        except KeyError as err:
+            raise MetaColumnNotFound(
+                f"{name_column} column with sheet is not found"
+            ) from err
+        worksheet_settings.setdefault(sheet_name, item)
```

Comment:
```suggestion
        if sheet_name in worksheet_settings:
            raise MetaColumnNotUnique(f"Duplicate sheet name {sheet_name} in settings meta column {name_column}")
        worksheet_settings[sheet_name] = item
```

The setdefault would simply do nothing instead of raising an error. Could also raise a ValueError if you don't want to introduce a new exception class.